### DISCUSSION TITLE
ente Desktop: add ability to open images with external programs

### DISCUSSION THREAD

**Discussion by mapperfr** [ente-io/ente#1720] [Enhancements]
The title says it all! It would be great to be able to easily open a picture with an external program (e.g. Darktable or another photo editor). 

Bonus functionality: detect image modifications and suggest to replace the old one by the new one when modified through an external program. (This is probably harder to implement but dream big, they say :) )

**Comment by oliversturm** [comment]
Yes... and on my Mac there are already apps registered as photo editors in Apple Photos. Alternatively all we need is the typical old "external tool" configuration, basically the name of an executable should be enough for the purpose.

I asked about this in Discord and saw a suggestion to create a discussion thread -- well here it is already.

So. Shouldn't be hard to do, right?

- Download image to temp location for editing
- Calculate hash value of image content
- Run external editor, i.e. `configuredExecutableFromAbove /path/to/temp/image.ext`, observe resulting process
- When image editor process ends, calculate new content hash, compare
- Re-upload image if changed compared to original image, either using original id or as new clone
- Remove temp file

There are lots of variations and optional feature ideas here, but I'd personally be quite happy with this. Thoughts?
